File: backend/trivia-data/src/main/java/com/delacruz/trivia/model/Room.java
package com.ectrvia.trivia.model;

import lombok.Data;
import java.time.LocalDateTime;
import java.util.List;

@Data
public class Room {
    private Long id;
    private String roomCode;
    private Long hostPlayerId;
    private Category category;
    private String status;
    private Boolean isThemeBased;
    private Integer questionTimerSeconds;
    private Integer maxPlayers;
    private Integer currentQuestionIndex;
    private Integer totalQuestions;
    private Integer playerCount;
    private List<Player> players;
    private LocalDateTime createdAt;
    private LocalDateTime startedAt;
    private LocalDateTime finishedAt;
}

File: ECTriviaApp/app/src/main/java/com/delacruz/ectrivia/data/model/Room.kt
package com.ectrvia.ectrivia.data.model

data class Room(
    val id: Long,
    val roomCode: String,
    val status: RoomStatus,
    val category: Category?,
    val isThemeBased: Boolean,
    val questionTimerSeconds: Int,
    val maxPlayers: Int,
    val currentQuestionIndex: Int,
    val totalQuestions: Int,
    val players: List<Player>,
    val createdAt: String
)

enum class RoomStatus {
    LOBBY,
    IN_PROGRESS,
    FINISHED,
    CANCELLED
}
